src/app/core/pagination/pagination.service.ts

```diff
@@ -0,0 +1,244 @@
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { RouteService } from '../services/route.service';
+import { PaginationComponentOptions } from '../../shared/pagination/pagination-component-options.model';
+import { combineLatest as observableCombineLatest, Observable } from 'rxjs';
+import { filter, map, take } from 'rxjs/operators';
+import { SortDirection, SortOptions } from '../cache/models/sort-options.model';
+import { FindListOptions } from '../data/request.models';
+import { hasValue, isEmpty, isNotEmpty } from '../../shared/empty.util';
+import { difference } from '../../shared/object.util';
+import { isNumeric } from 'rxjs/internal-compatibility';
+
+
+@Injectable({
+  providedIn: 'root',
+})
+/**
+ * Service to manage the pagination of different components
+ * The pagination information will be stored in the route based on a paginationID.
+ * The following params are used for the different kind of pagination information:
+ *    - For the page: p.{paginationID}
+ *    - For the page size: rpp.{paginationID}
+ *    - For the sort direction: sd.{paginationID}
+ *    - For the sort field: sf.{paginationID}
```

Comment:
Tiny thing...this comment might be updated for the new format, e.g. `{paginationID}.page`, etc.